src/server/routes/csv.js

```diff
@@ -67,9 +67,13 @@ router.use(function (req, res, next) {
 					(await isTokenAuthorized(token, csvRole)) ? cb(null, true) : cb(new Error('Invalid token (either unauthorized or logged out'));
 				} else {
 					// If no token is found, then the request is mostly like a curl request. We require an
-					// email and password to be supplied for curl requests.
-					const { email, password } = request.body;
-					const verifiedUser = await verifyCredentials(email, password, true);
+					// username and password to be supplied for curl requests.
+					const { username, email, password } = request.body;
+					// allowing for backwards compatibility if previous eGuage meters are using the 'email' parameter instead of
```

Comment:
Same thoughts as other places it was changed.

Reply:
Same as others, added TODO, username parameter is taken if it is given, otherise email parameter is taken as username. If both are supplied, username takes priority. If niether is given, then fails.